11-lorawan/test_spec11.py

```diff
@@ -0,0 +1,118 @@
+import time
+import pytest
+from testutils.shell import GNRCLoRaWANSend, ifconfig, lorawan_netif
+from riotctrl_shell.netif import Ifconfig
+import base64
+import json
+
+APP = 'examples/gnrc_lorawan'
+pytestmark = pytest.mark.rc_only()
+
+APP_PAYLOAD = "This is RIOT!"
+DOWNLINK_PAYLOAD = "VGhpcyBpcyBSSU9U"
+APP_PORT = 2
+RX2_DR = 3
+
+
+class Shell(Ifconfig, GNRCLoRaWANSend):
+    pass
+
+
+def run_lw_test(node, client, userdata, iface, app_id, dev_id):
+    # Disable confirmable messages
+    node.ifconfig_flag(iface, "ack_req", enable=False)
+
+    # Push a downlink message to the TTN server
+    dl_data = {"payload_raw": DOWNLINK_PAYLOAD, "port": APP_PORT,
+               "confirmed": True}
+
+    client.publish("{}/devices/{}/down".format(app_id, dev_id),
+                   json.dumps(dl_data))
```

Comment:
Make this a method of `testutils.ttn.TTNClient`:

```py
def publish(app_id, dev_id, **kwargs):
    self.client.publish("{}/devices/{}/down".format(app_id, dev_id),
                        **kwargs))
```

Reply:
Arghs sorry, the `self.client.publish()` call has to look like this of course:

```py
    self.client.publish("{}/devices/{}/down".format(app_id, dev_id),
                        json.dumps(kwargs))
```

Reply:
And maybe call it `publish_to_dev` or something instead of just `publish`.

Reply:
done!